pandas/core/groupby/generic.py

```diff
@@ -187,6 +192,31 @@ def pinner(cls):
     return pinner
 
 
+def _cast_agg_result(
```

Comment:
This function is cut and paste from `_cython_agg_blocks` below (it was currently defined within the method). The only thing I added is a `reshape1d` keyword to prevent reshaping 1d numpy arrays to 2D when using it for the ArrayManager

Reply:
can you just add it to the original?